lib/rucio/core/credential.py

```diff
@@ -107,4 +116,43 @@ def get_signed_url(service, operation, url, lifetime=600):
                                                                                                      lifetime,
                                                                                                      signature)
 
+    elif service == 's3':
+        # S3 can't sign delete URLs
+        if operation == 'delete':
+            raise UnsupportedOperation('Delete operation not supported for S3 signed URLs')
+
+        # split URL to get hostname, bucket and key
+        components = url.split('/')
```

Comment:
maybe this is better done with `urllib.parse` ?

Reply:
I have modified the code to use `urllib.parse`.